MetaLamp/nft-marketplace/src/Plutus/Contracts/Services/Sale/StateMachine.hs

```diff
@@ -71,18 +72,20 @@ transition :: GetAdditionalConstraints -> Sale -> State SaleDatum -> SaleRedeeme
 transition additionalConstraints sale@Sale{..} state redeemer = case (stateData state, redeemer) of
     (SaleOngoing, Redeem)
         -> Just ( Constraints.mustBeSignedBy saleOwner <>
-                  Constraints.mustPayToPubKey saleOwner val
+                  Constraints.mustPayToPubKey saleOwner saleValueWithMinAdaTxOut
                 , State SaleClosed mempty
                 )
-    (SaleOngoing, Buy buyer) | saleValue == val
+    (SaleOngoing, Buy buyer) | saleValueWithMinAdaTxOut == val
         -> Just ( Constraints.mustBeSignedBy buyer <>
-                  Constraints.mustPayToPubKey buyer saleValue <>
+                  Constraints.mustPayToPubKey buyer saleValueWithMinAdaTxOut <>
+                  -- TODO: is it okay that buyer receive additional 2ADA? Should we initially add them to the sale price?
```

Comment:
In my opinion saler (`sfSaleOperator`) should receive this 2 ADA back as it was his value locked initially